src/uct/base/uct_md.c

```diff
@@ -249,26 +249,23 @@ static ucs_status_t uct_config_read(uct_config_bundle_t **bundle,
     return status;
 }
 
-uct_tl_component_t* uct_md_get_single_tlc(uct_md_h md)
-{
-    uct_md_registered_tl_t *tlr;
-
-    ucs_assert_always(ucs_list_length(&md->component->tl_list) == 1);
-    tlr = ucs_list_head(&md->component->tl_list, uct_md_registered_tl_t, list);
-    return tlr->tl;
-}
-
 static uct_tl_component_t *uct_find_tl_on_md(uct_md_component_t *mdc,
                                              const char *tl_name)
 {
     uct_md_registered_tl_t *tlr;
 
-    ucs_list_for_each(tlr, &mdc->tl_list, list) {
-        if (!strcmp(tl_name, tlr->tl->name)) {
-            return tlr->tl;
+    if (tl_name == NULL) {
+        ucs_assert_always(ucs_list_length(&mdc->tl_list) == 1);
+        tlr = ucs_list_head(&mdc->tl_list, uct_md_registered_tl_t, list);
```

Comment:
this function should return NULL if not found, and not assert. since if no tl is found, it means the user has provided invalid argument, and uct_md_iface_config_read() should return UCS_ERR_NO_DEVICE, instead of failing on assertion. just like in regular transport case.

Reply:
ok, can keep the assert? (not assert_always)
also, it's not exactly that the user provided an invalid argument, becuase he set tl_name to NULL. it means that there is no tl registered on the md and it's not the user's fault...

Reply:
it means the user has passed md which does not have sockaddr - which is invalid usage